### PR TITLE
BaSP-TG-6: Projects Page - Add/Edit Form

### PR DESCRIPTION
**Pull request by DamianPalavecino**
- Create add/edit form including: Project Name, Description, Start Date, End Date, Client Name
- Redirect to Projects List (Name - Description - Client - Start Date- End Date - Employees - Actions)
- On add functionality the default value for Status is "inactive"
- On edit functionality should show previuos info
- Success message for confirmation edit and add actions
- Redirect to add/edit form page
- Update Project Status in create/edit form
- When creating a project, status is set to inactive

### CODE REVIEW

File: src/Components/Projects/modal.module.css [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    border-radius: 5px;
+    color: white;
+    margin-left: 10px;
+}
+
+.buttonClose {
+    padding: 10px 24px;
+    background: #AACE9B;
+    border-radius: 5px;
+    color: white;
```
Comment (mara666) [inline src/Components/Projects/modal.module.css:80]: colors must be in hexa!

File: src/Components/Projects/projectForm.jsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    setInputValue({ ...inputValue, [e.target.name]: e.target.value });
+  };
+
+  const redirect = () => {
+    window.location.assign('/projects');
+  };
+
+  const onSubmit = () => {
+    if (id) {
+      console.log(inputValue);
```
Comment (mara666) [inline src/Components/Projects/projectForm.jsx:47]: please, delete this console.log()

File: src/Components/Projects/projectForm.jsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(inputValue)
+      };
+      fetch(url, post).then(async (response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          const { message } = await response.json();
+          alert(message);
+        } else {
+          alert('Project was created successfully');
```
Comment (mara666) [inline src/Components/Projects/projectForm.jsx:78]: here you can also print message from BE, like line 76

File: src/Components/Projects/projectForm.jsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        },
+        body: JSON.stringify(inputValue)
+      };
+      console.log(JSON.stringify(inputValue));
+      fetch(url + '/' + id, put).then(async (response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          const { message } = await response.json();
+          alert(message);
+        } else {
+          alert('Project was successfully edited');
```
Comment (mara666) [inline src/Components/Projects/projectForm.jsx:61]: here you can also print message from BE, like line 59

File: src/Components/Projects/projectForm.jsx (hunk 3 above)
Comment (martumoser) [reply src/Components/Projects/projectForm.jsx:78]: I'm not sure that the response of the endpoint includes a message. Check that before do the changes

File: src/Components/Projects/modal.module.css (hunk 1 above)
Comment (DamianPalavecino) [reply src/Components/Projects/modal.module.css:80]: Done

File: src/Components/Projects/projectForm.jsx (hunk 2 above)
Comment (DamianPalavecino) [reply src/Components/Projects/projectForm.jsx:47]: Done

File: src/Components/Projects/projectForm.jsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    description: '',
+    clientName: ''
+  });
+  const fixDate = (date) => {
+    return date.slice(0, 10);
+  };
+
+  if (id) {
+    useEffect(async () => {
+      if (id) {
```
Comment (Juanoauil) [inline src/Components/Projects/projectForm.jsx:21]: why do you ask twice for the id

File: src/Components/Projects/projectForm.jsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    if (id) {
+      console.log(inputValue);
+      const put = {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(inputValue)
+      };
+      console.log(JSON.stringify(inputValue));
```
Comment (Juanoauil) [inline src/Components/Projects/projectForm.jsx:55]: please remove console.log

**Review by Juanoauil — CHANGES_REQUESTED** [review]

File: src/Components/Projects/projectForm.jsx (hunk 3 above)
Comment (DamianPalavecino) [reply src/Components/Projects/projectForm.jsx:78]: Done

File: src/Components/Projects/projectForm.jsx (hunk 6 above)
Comment (DamianPalavecino) [reply src/Components/Projects/projectForm.jsx:55]: Done

File: src/Components/Projects/projectForm.jsx (hunk 5 above)
Comment (DamianPalavecino) [reply src/Components/Projects/projectForm.jsx:21]: Mistake, fixed

File: src/Components/Projects/projectForm.jsx (hunk 4 above)
Comment (DamianPalavecino) [reply src/Components/Projects/projectForm.jsx:61]: Done

File: src/Components/Projects/index.jsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+          <Modal
+            project={modalObject}
+            handleDelete={handleDelete}
+            type={modalType}
+            showModal={showModal}
+            closeModal={closeModal}
+          />
+          {projects === undefined && <p>There are no projects</p>}
+          {projects !== undefined &&
+            projects.map((project) => {
```
[the remark this answers is not in the document]
Comment (DamianPalavecino) [reply src/Components/Projects/index.jsx:68]: Done

**Review by Juanoauil — APPROVED** [review]

**Review by mara666 — APPROVED** [review]
Good job Dami!

**Comment by cami-diaz** [comment]
I can edit a project and save it without filling in the customer name. But in the projects list it keeps coming out with the old data.
image

**Comment by cami-diaz** [comment]
Employees added to the project cannot be viewed. There is also no option in the form to add employee when adding or editing a project.

image

**Pull request merged**